src/sentry/workflow_engine/detectors/error.py

```diff
@@ -0,0 +1,171 @@
+from collections.abc import MutableMapping
+from dataclasses import dataclass
+from datetime import datetime, timedelta, timezone
+from enum import StrEnum
+from time import time
+from typing import TYPE_CHECKING, Any, TypedDict, TypeVar
+
+from django.utils import timezone as django_timezone
+
+from sentry import analytics
+from sentry.event_manager import assign_event_to_group
+from sentry.integrations.tasks.kick_off_status_syncs import kick_off_status_syncs
+from sentry.issues import grouptype
+from sentry.metrics.middleware import MutableTags
+from sentry.models.activity import Activity
+from sentry.models.group import Group, GroupStatus
+from sentry.models.grouphistory import GroupHistoryStatus, record_group_history
+from sentry.models.groupinbox import GroupInboxRemoveAction, remove_group_from_inbox
+from sentry.models.options.project_option import ProjectOption
+from sentry.models.project import Project
+from sentry.signals import issue_resolved
+from sentry.types.activity import ActivityType
+from sentry.workflow_engine.models import DataPacket
+from sentry.workflow_engine.models.data_condition import Condition, condition_registry
+from sentry.workflow_engine.processors.detector import DetectorEvaluationResult, DetectorHandler
+from sentry.workflow_engine.types import DetectorGroupKey
+
+if TYPE_CHECKING:
+    from sentry.eventstore.models import Event
+
+T = TypeVar("T")
+
+
+class ErrorDetectorConfigType(StrEnum):
+    GROUPING = "ErrorGroupingCondition"
+    RESOLVE = "ErrorResolveCondition"
+
+
+class ErrorDetectorHandler(DetectorHandler[T]):
+    def evaluate(
+        self, data_packet: DataPacket[T]
+    ) -> dict[DetectorGroupKey, DetectorEvaluationResult]:
+        pass
+
+    def apply(self, type: ErrorDetectorConfigType, data: dict[str, Any]):
+        assert self.condition_group
+        config = self.condition_group.conditions.get(
+            type=type
+        )  # unused condition model for project options. can migrate later, but make sure it exists
```

Comment:
how do we make it easy when we add new conditions that we SHOULD evaluate? maybe the condition class should take the `DataCondition` as a parameter and you can use it if you want

Reply:
i think we should still make the DataConditions via the API for these. serialization is hard though, since it doesn't map neatly to the comparison for a DataCondition. here, all the `DataCondition.type` does is tell you to get the `Condition` dataclass which has all the logic.

it's not clear to me if we need to store grouping rules in DataConditions since they're already nicely stored as ProjectOptions, are very long, and span multiple options.

for auto-resolve it's also made up of 2 options, one of which stores the last time we batched an auto-resolve task, this would be frequently updated as a `DataCondition`